src/plots/cartesian/axes.js

```diff
@@ -944,9 +952,61 @@ axes.calcTicks = function calcTicks(ax, opts) {
             axes.prepTicks(mockAx, opts);
         }
 
+        if(mockAx.tickmode === 'full domain') {
+            var nt = mockAx.nticks;  // does mockAx have nitkcs?
+            if(nt === undefined) nt = 0;
+            if(nt === 0) {
+              // pass
+            } else if(nt === 1) {
+                tickVals = [0];
+            } else if(nt === 2) {
+                tickVals = [0, 1];
+            } else {
+                var increment = 1 / (nt - 1); // (nt-2) + 1
+                tickVals.push(0);
+                for(var tickIndex = 0; tickIndex < nt - 2; tickIndex++) {
+                    tickVals.push((tickIndex + 1) * increment);
+                }
+                tickVals.push(1);
```

Comment:
For major ticks this is good.

For minor, with the existing `'tickmode: 'auto'` we take the calculated major ticks, divide the intervals by `minor.nticks`, and draw minor ticks everywhere _except_ where there are already major ticks. Which means with `minor.nticks=5` we actually draw 4 minor ticks in each major interval. That's a little confusing compared with what we're doing here with the new `'full domain'` major ticks but I don't see a better design so let's keep major as is, and have minor behave in the analogous way: when major and minor are both `'full domain'` we should draw `minor.nticks - 1` ticks in each major tick interval.

What about other major/minor tickmode combinations? I think it'd be reasonable to restrict the allowed `minor.tickmode` based on major `tickmode` so that at least the major and minor ticks always move together:
- `tickmode` in `'auto', 'linear', 'array'` -> `minor.tickmode` also in `'auto', 'linear', 'array'`. We currently allow `tickmode: 'array', minor.tickmode: 'auto'` but this seems like a bug to me as the minor ticks don't (and can't!) evenly divide major tick intervals, and the spacing of minor ticks depends on `minor.nticks` but in some strange way. @archmoj was this intentional or should we prohibit `minor.tickmode: 'auto'` with `tickmode: 'array'`?
- `tickmode: 'sync'` -> minor ticks should match the minor ticks on the sync'd axis, which they currently almost do, but we get some extras so this is a bug (cc @archmoj - try opening the `new_tickmode_sync` mock and calling `Plotly.relayout(gd,{'yaxis2.minor.ticks':'outside'})` - you'll see the correct minor ticks plus 11 extras, some of which coincide with the expected ticks, some don't, you can see them all if you drag only one of the y axes). Seems to me you should be able to set the visibility and appearance of minor ticks/grid on this axis, but their positions should be sync'd to the positions of minor ticks on the main axis
- `tickmode` in `'full domain', 'domain array'` -> `minor.tickmode` also in `'full domain', 'domain array'`. But per my comment above I think with `tickmode: 'domain array'` we should only accept `minor.tickmode: 'domain array'` too, or no minor ticks.

Reply:
I'll take a look at getting minor ticks spaced out between major ticks.

Does plotly throw/raise errors on invalid combinations, or is silent?